spec/unit/webrtc/callFeed.spec.ts

```diff
@@ -85,5 +85,129 @@ describe("CallFeed", () => {
                 expect(feed.isVideoMuted()).toBeTruthy();
             });
         });
+
+        describe("voice activity detection", () => {
+            it("voice activity should disable audio track", () => {
+                feed.stream.addTrack(
+                    //@ts-ignore Mock
+                    new MockMediaStreamTrack("track", "audio", true),
+                );
+
+                feed.setVoiceActivityThreshold(Infinity);
+
+                setTimeout(() => {
+                    expect(feed.stream.getAudioTracks()[0].enabled).toBe(false);
+                }, 1000);
+            });
+
+            it("voice activity should enable audio track", () => {
+                feed.stream.addTrack(
+                    //@ts-ignore Mock
+                    new MockMediaStreamTrack("track", "audio", true),
+                );
+
+                feed.setVoiceActivityThreshold(-Infinity);
+                setTimeout(() => {
+                    expect(feed.stream.getAudioTracks()[0].enabled).toBe(true);
+                }, 1000);
+            });
+
+            it("voice activity should enable audio track", () => {
+                feed.stream.addTrack(
+                    //@ts-ignore Mock
+                    new MockMediaStreamTrack("track", "audio", true),
+                );
+
+                feed.setVoiceActivityThreshold(-50);
+                feed.speakingVolumeSamples = [-40];
+
+                setTimeout(() => {
+                    expect(feed.stream.getAudioTracks()[0].enabled).toBe(true);
+                }, 1000);
+            });
+
+            it("voice activity should disable audio track", () => {
+                feed.stream.addTrack(
+                    //@ts-ignore Mock
+                    new MockMediaStreamTrack("track", "audio", true),
+                );
+
+                feed.setVoiceActivityThreshold(-50);
+                feed.speakingVolumeSamples = [-60];
+
+                setTimeout(() => {
+                    expect(feed.stream.getAudioTracks()[0].enabled).toBe(false);
+                }, 1000);
+            });
+
+            it("voice activity should not disable audio track after a few milliseconds", async () => {
```

Comment:
To be completely honest, it's tough for me to follow what part of the cooldown tests does what (i.e. start/stop speaking) - could you please add comments and also avoid leaving things out that don't need to be in the test but would be better if they were there to be explicit (if I do understand everything correctly, you're not adding samples for all the start/stop speaking changes?)

Reply:
I added some comments, they are very likely a bit too much but should at least give a better picture, I can then remove the superfluous ones later.

With leaving things out, is there something we can combine here? Like only create one stream?
Not sure what exactly you meant with that.